src/dtclient/processmoduleconfig.go

```diff
@@ -21,33 +21,64 @@ type ProcessModuleProperty struct {
 	Value   string `json:"value"`
 }
 
-func (pmc *ProcessModuleConfig) AddHostGroup(hostGroup string) *ProcessModuleConfig {
+func (pmc *ProcessModuleConfig) Add(newProperty ProcessModuleProperty) *ProcessModuleConfig {
 	if pmc == nil {
 		pmc = &ProcessModuleConfig{}
 	}
-	newProps := []ProcessModuleProperty{}
-	hasHostGroup := false
-	for _, prop := range pmc.Properties {
-		if prop.Key != "hostGroup" {
-			newProps = append(newProps, prop)
+
+	var newProps []ProcessModuleProperty
+	hasPropertyGroup := false
+	for _, currentProperty := range pmc.Properties {
+		if currentProperty.Key != newProperty.Key {
+			newProps = append(newProps, currentProperty)
 		} else {
-			hasHostGroup = true
-			if hostGroup == "" {
+			hasPropertyGroup = true
+			if newProperty.Value == "" {
 				continue
-			} else if hostGroup == prop.Value {
-				newProps = append(newProps, prop)
+			} else if newProperty.Value == currentProperty.Value {
+				newProps = append(newProps, currentProperty)
 			} else {
-				newProps = append(pmc.Properties, ProcessModuleProperty{Section: "general", Key: "hostGroup", Value: hostGroup})
+				newProps = append(pmc.Properties, currentProperty)
 			}
 		}
 	}
-	if !hasHostGroup && hostGroup != "" {
-		newProps = append(pmc.Properties, ProcessModuleProperty{Section: "general", Key: "hostGroup", Value: hostGroup})
+	if !hasPropertyGroup && newProperty.Value != "" {
+		newProps = append(pmc.Properties, newProperty)
 	}
 	pmc.Properties = newProps
 	return pmc
 }
 
+func (pmc *ProcessModuleConfig) AddConnectionInfo(connectionInfo ConnectionInfo) *ProcessModuleConfig {
+	tenant := ProcessModuleProperty{
+		Section: "general",
```

Comment:
using `const`s would actually be nice improvement (you followed my example, so I feel a bit guilty 😅 )

Reply:
Following "rule of three", made "general" a const and left single-use hardcoded string as they are.